swap/swap_test.go

```diff
@@ -682,3 +686,256 @@ func testDeploy(ctx context.Context, backend cswap.Backend, swap *Swap) (err err
 	}
 	return nil
 }
+
+// TestSaveAndLoadLastReceivedCheque tests if a saved last received cheque can be loaded again later using the swap functions
+func TestSaveAndLoadLastReceivedCheque(t *testing.T) {
+	swap, dir := newTestSwap(t)
+	defer os.RemoveAll(dir)
+
+	testID := newDummyPeer().Peer.ID()
+	testCheque := newTestCheque()
+
+	if err := swap.saveLastReceivedCheque(testID, testCheque); err != nil {
+		t.Fatalf("Error while saving: %s", err.Error())
+	}
+
+	returnedCheque := swap.loadLastReceivedCheque(testID)
+
+	if returnedCheque == nil {
+		t.Fatalf("Could not find saved cheque")
+	}
+
+	if returnedCheque.Amount != testCheque.Amount || returnedCheque.Beneficiary != testCheque.Beneficiary {
```

Comment:
Is there any reason why you are checking only these 2 fields, rather than all of them? (I was thinking of something like an _equals_ method)

Reply:
Mainly because there is no equals function yet. The idea behind these 2 fields was that if those 2 were set correctly the rest most likely was as well.

Reply:
I'm actually working on a separate PR where I extract the cheque related functions from swap into their own package. There I will add an equal and then update the tests.